src/audio/drc/drc.c

```diff
@@ -304,189 +301,190 @@ static int drc_trigger(struct comp_dev *dev, int cmd)
 
 	return ret;
 }
+#endif
 
-static void drc_process(struct comp_dev *dev, struct comp_buffer __sparse_cache *source,
-			struct comp_buffer __sparse_cache *sink, int frames,
-			uint32_t source_bytes, uint32_t sink_bytes)
+static int drc_set_config(struct processing_module *mod, uint32_t config_id,
+			  enum module_cfg_fragment_position pos, uint32_t data_offset_size,
+			  const uint8_t *fragment, size_t fragment_size, uint8_t *response,
+			  size_t response_size)
 {
-	struct drc_comp_data *cd = comp_get_drvdata(dev);
+	struct drc_comp_data *cd = module_get_private_data(mod);
 
-	buffer_stream_invalidate(source, source_bytes);
+	comp_info(mod->dev, "drc_set_config()");
 
-	cd->drc_func(dev, &source->stream, &sink->stream, frames);
+	return comp_data_blob_set(cd->model_handler, pos, data_offset_size, fragment,
+				  fragment_size);
+}
+
+static int drc_get_config(struct processing_module *mod,
+			  uint32_t config_id, uint32_t *data_offset_size,
+			  uint8_t *fragment, size_t fragment_size)
+{
+	struct sof_ipc_ctrl_data *cdata = (struct sof_ipc_ctrl_data *)fragment;
+	struct drc_comp_data *cd = module_get_private_data(mod);
 
-	buffer_stream_writeback(sink, sink_bytes);
+	comp_info(mod->dev, "drc_get_config()");
 
-	/* calc new free and available */
-	comp_update_buffer_consume(source, source_bytes);
-	comp_update_buffer_produce(sink, sink_bytes);
+	return comp_data_blob_get_cmd(cd->model_handler, cdata, fragment_size);
 }
 
-/* copy and process stream data from source to sink buffers */
-static int drc_copy(struct comp_dev *dev)
+static void drc_set_alignment(struct audio_stream __sparse_cache *source,
+			      struct audio_stream __sparse_cache *sink)
 {
-	struct comp_copy_limits cl;
-	struct drc_comp_data *cd = comp_get_drvdata(dev);
-	struct comp_buffer *sourceb, *sinkb;
-	struct comp_buffer __sparse_cache *source_c, *sink_c;
-	int ret = 0;
+	const uint32_t byte_align = 1;
+	const uint32_t frame_align_req = 1;
 
-	comp_dbg(dev, "drc_copy()");
+	audio_stream_init_alignment_constants(byte_align, frame_align_req, source);
```

Comment:
possible just pass 1 as parameter and add comments to say the reason for set alignment to 1?